pokemon_battle_arena/src/arena/mod.rs

```diff
@@ -49,10 +52,12 @@ impl Arena {
     // Setter Methods
     //
     /// Sets the effect of the arena
-    pub fn set_effect(&mut self, new: enums::types) {
+    #[allow(dead_code)]
+    pub fn set_effect(&mut self, new: enums::Types) {

```

Comment:
But since this is not a 1.0 API which is fixed for eternity, they can change access to the field to setter/getter later. But ok, I agree: in case this is really not a POD type (I haven't looked at the definition, commenting took long enough already) and it's not unlikely that checks are needed in the future, it's not a bad idea to keep the setter/getter pairs.